Dynamic metatgas and builder page OG image for easy sharing on x, farcaster.

NikolaiL opened the pull request:
This PR adds dynamic metatags and dynamically generated builder images for sharing.


Image examples:
 
<img width="1200" height="800" alt="ogpablo" src="https://github.com/user-attachments/assets/775e6ce9-9750-4e67-8507-d0b93e891546" />

<img width="1200" height="800" alt="ogbhavyagor12" src="https://github.com/user-attachments/assets/05aa283e-2084-46d3-a4de-d79ffa99b9ec" />

(I left some vertical margin so it renders nicely on x, farcaster and other platforms, as all of them have slightly different image size requirements.

metatags:

<img width="673" height="67" alt="Screenshot 2025-10-06 at 16 00 47" src="https://github.com/user-attachments/assets/b0a518e5-3519-4dd9-a008-43cc0c257b2c" />


Please let me know what additional information do we need on the images / metatags.

Once it is approved I can start add metatags needed to make it a render as a mini app on farcaster.

bhavyagor12 commented:
Have some suggestions with the data that is going to be displayed when the link is shared, specially around what we could display for a non dev creds user to understand what the fields - "cred score", "attestations" and "verified attestations" 

Feel like one score that aggregates everything is better for an user to understand and then we can have them come into the platform to get to know more
I feel when a developer shares their builder profile - they would what to show more things like skills and how many verifications they got for the respective skill


Open to discussion, really nice work @NikolaiL !!!!

NikolaiL commented:
> skills and how many verifications they got for the respective skill

Will add this.

> Open to discussion, really nice work @NikolaiL !!!!

Thank you!

NikolaiL commented:
Added skills, names, bio, and a fallback if no name is present.

Here is an example with name present:

<img width="1200" height="800" alt="image" src="https://github.com/user-attachments/assets/fb81530e-98f5-439e-8879-34661fa8c1fb" />


and with name missing:

<img width="1200" height="800" alt="image" src="https://github.com/user-attachments/assets/fe4b24e6-c0d5-4573-9567-6b4feaa9526f" />

Pabl0cks reviewed (commented):
Good job @NikolaiL!!! 🔥🔥

I really like the og image, and thanks @bhavyagor12 for the suggestions, It’s looking clean with just the score and the skills 🙏

A couple of suggestions:

- Update the title of the page to something along these lines: `username Developer Attested Skills | DevCreds` or `username Developer Attested Credencials | DevCreds`. Current title `username | DevCreds | Scaffold-ETH 2` is too generic.

- With the new simplified layout, there’s some empty space below the dev description. Maybe we could slightly reduce the PFP size and move the CTA to the top of the card with a slightly larger font. Quick and dirty test:

<img width="1200" height="800" alt="image" src="https://github.com/user-attachments/assets/6e3d123d-129b-46fc-adcf-bb7bfc34b157" />

NikolaiL commented:
@Pabl0cks , thank you for comments!

Updated page page titles:

<img width="564" height="117" alt="Screenshot 2025-10-07 at 20 56 28" src="https://github.com/user-attachments/assets/e4b96aed-02c9-4e93-bc9f-cae7042855e2" />


Updated Image:

<img width="1200" height="800" alt="image" src="https://github.com/user-attachments/assets/145d0998-f525-466d-88bc-dbae37828960" />

For now I left the CTA on the bottom, but made the font bigger. Let me know if you want it more big.

bhavyagor12 commented:
I love this @NikolaiL !!!!!

LGTM!! 
@Pabl0cks you can have a look and lets ship this !!!
